system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj

```diff
@@ -346,3 +350,57 @@
                :echo10 (search/find-metadata :collection :echo10 {:include-has-granules true :include-granule-counts true})
                :atom (search/find-concepts-atom :collection {:include-has-granules true :include-granule-counts true})
                :atom (search/find-concepts-json :collection {:include-has-granules true :include-granule-counts true})))))))
+
+(deftest search-collections-has-granules-or-cwic-test
+  (let [coll1 (make-coll 1 m/whole-world nil)
+        coll2 (make-coll 2 m/whole-world nil)
+        coll3 (make-coll 3 m/whole-world nil)
+        coll4 (make-coll 4 m/whole-world nil)
+        coll5 (make-coll 5 m/whole-world nil)
+        coll6 (make-coll 6 m/whole-world nil)
+
+        _ (index/wait-until-indexed)
+        user1-token (e/login (s/context) "user1")
+        tag1 (tags/save-tag
+               user1-token
+               (tags/make-tag {:tag-key "NON-CWIC"})
+               [coll1 coll5])
+        tag2 (tags/save-tag
+               user1-token
+               (tags/make-tag {:tag-key "org.ceos.wgiss.cwic.granules.prod"})
+               [coll2 coll6])]
+    (index/wait-until-indexed)
+
+    ;; coll1
+    ;; non-cwic tagged with granule
+    (make-gran coll1 (p/point 0 0) nil)
+
+    ;; coll2
+    ;; cwic tagged with granule
+    (make-gran coll2 (p/point 0 0) nil)
+
+    ;; coll 3
+    ;; no tag with granule
+    (make-gran coll3 (p/point 0 0) nil)
+
+    ;; coll 4
+    ;; no tag without granule
+
+    ;; coll 5
+    ;; non-cwic tagged no granule
+
+    ;; coll 6
+    ;; cwic tagged no granule
+
+    (index/wait-until-indexed)
+    (testing "Search with has-granules-or-cwic feature"
+      (d/refs-match? [coll1 coll3 coll6 coll2]
+                     (search/find-refs :collection
+                                       {:has_granules_or_cwic true}
+                                       {:snake-kebab? false})))
```

Comment:
The Acceptance criteria says "Upon selecting the "Only include collections with granules" checkbox, search results display collections with granules in the CMR and collections with granules that have CWIC tags."  
This doesn't sound right. Could you update it?

Your implementation is to display collections with granules and collections with CWIC tags, which is consistent with the the ticket description.

Reply:
Yea you are right. This needs to change to be more relevant to CMR.